### PR TITLE
Migrate package to TypeScript

### PR DESCRIPTION
**Pull request by filipsobol**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))  

Feature: Migrate package to Typescript. Closes [#13543](https://github.com/ckeditor/ckeditor5/issues/13543).

### CODE REVIEW

**Review by psmyrek — COMMENTED** [review]
The changes work fine in the Vue 3+ app.

Unfortunately, the changes do not work as expected in the Vue 2.x. After adding this new TypeScript-based `ckeditor5-vue` component into Vue 2.x app, we should see something like this in the console:

before

Instead, we get just a strange error:

after

Would it be possible to restore the previous behavior?

Another issue I found: `yarn run test` or `yarn run coverage` (or the `npm` equivalent) ends with an error:

err

Both commands works fine for me on `#master`. Any ideas?

File: tsconfig.json
[the remark this answers is not in the document]
Comment (filipsobol) [reply tsconfig.json:1]: I generated this fie using `tsc --init`. I personally like to keep the commented out lines to not have to check the docs when I need to add some property in the future.

If this is an issue, I can remove those lines.

File: yarn.lock
[the remark this answers is not in the document]
Comment (filipsobol) [reply yarn.lock:1]: I wanted to make it consistent with Travis config, which [uses npm](https://github.com/ckeditor/ckeditor5-vue/blob/master/.travis.yml#L16-L19). I feel like using `yarn` in this repo doesn't bring much, does it?

File: tsconfig.json
[the remark this answers is not in the document]
Comment (psmyrek) [reply tsconfig.json:1]: I removed the commented out lines.

File: yarn.lock
[the remark this answers is not in the document]
Comment (psmyrek) [reply yarn.lock:1]: Makes sense, thanks.

**Review by psmyrek — APPROVED** [review]
LGTM 🎉 

I extracted the issue with failing test commands on Windows (`npm run test` and `npm run coverage`) to #241 to not block the migration to TypeScript.

**Pull request merged**